### PR TITLE
Fix compiler crash when binding missing property pattern member.

### PR DESCRIPTION
**Pull request by gafter**
@agocke @cston @dotnet/roslyn-compiler Please review this tiny pattern-matching bug fix.

@CyrusNajmabadi This should fix a crash you ran into in #10873

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Awesome.  Thanks!

File: src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs [hunk 1]
```diff
@@ -658,14 +658,16 @@ private BoundPattern BindPropertyPattern(PropertyPatternSyntax node, TypeSymbol
         {
             Symbol symbol = BindPropertyPatternMember(inputType, name, ref hasErrors, diagnostics);
 
-            if (inputType.IsErrorType() || hasErrors)
+            if (inputType.IsErrorType() || hasErrors || symbol == (object)null)
```
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs]: consider ```is null```.

**Comment by CyrusNajmabadi** [comment]
@gafter  i have tried out your fix.  I do not get an error anymore.  however, i also do not get compiler errors in cases i had expected.  For example:

```c#
class C
{
    void M2()
    {
        object o = null;
        switch (o)
        {
            case Blah { X: int i }:
                break;
        }
    }

    class Blah
    {
    }
}
```

THere is no error about 'X' being undefined in that pattern.  Is that expected?

**Comment by gafter** [comment]
> There is no error about 'X' being undefined in that pattern. Is that expected?

No, but that is almost precisely one of the tests I added that shows an error being reported. Not sure why you're not seeing it.

**Comment by CyrusNajmabadi** [comment]
Yes.  I see the error with "is patterns".  I don't see any error with pattern switch cases.  Do you?

**Comment by gafter** [comment]
I've added tests with a switch statement to demonstrate those errors.

**Comment by CyrusNajmabadi** [comment]
>I've added tests with a switch statement to demonstrate those errors.

@gafter All those tests are with tuples/deconstruction.  Do you have any tests with a property pattern (like in the example i showed)?   I can even see this behavior in the IDE.  I get no errors for the example i provided.

**Comment by gafter** [comment]
These look like exactly the tests we need: https://github.com/dotnet/roslyn/pull/26073/commits/afc718817b891a4297d06a3e16da11a764cff397

**Pull request merged**